src/rebar_config.erl

```diff
@@ -239,11 +239,25 @@ consult_file_(File) ->
 -spec verify_config_format([{_,_}]) -> true.
 verify_config_format([]) ->
     true;
+verify_config_format([{_Key, Value} | T] ) when is_list(Value) ->
+    verify_config_format(T) andalso verify_config_format_sub(Value);
 verify_config_format([{_Key, _Value} | T]) ->
     verify_config_format(T);
 verify_config_format([Term | _]) ->
     throw(?PRV_ERROR({bad_config_format, Term})).
 
+%% @private checks that sub lists (in the top level key value pairs)
+%% do not contain empty, or size 1 tuples
+-spec verify_config_format_sub([tuple()]) -> true.
+verify_config_format_sub([]) ->
+    true;
+verify_config_format_sub([{_}=Term | _]) ->
+    throw(?PRV_ERROR({bad_config_format, Term}));
```

Comment:
hm, this is a tricky one.  I guess it has no reason to ever happen, and I believe no plugins currently use that format, so it shouldn't be a problem.

However, it could prevent any plugin from doing something like `{arbitrary_terms, [{a}]}` and having their config work.

It might catch more errors than it causes though. @tsloughter any opinions?